threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,84 @@
+- **File System**: The file system where Notation or application built with Notation library is installed. For Notation signing artifacts as OCI image layout, it is the place where artifacts and associated signatures are stored. There are different directories and files as follows:
+  - Notation config file, where general configuration is stored
+  - Notation trust policy file, where trust policy configuration is stored
+  - Notation trust store, where certificate files are stored to validate signatures
+  - Notation plugin directory, where Notation plugins are installed
+  - Credential store, where the registry or Key management credentials are stored
+- **Notation Plugin**: An external component built by different vendors, and integrated to Notation as a signing plugin for signature generation, or as a verifying plugin for signature verification, see [plugin spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/plugin-extensibility.md) for details.
+- **Registry**: A registry for Docker and Open Container Initiative (OCI) images, with support for all OCI artifacts. It is the storage where artifacts are pushed to by artifact builder. Registries are outside Notation trust boundary.
+- **KMS**: A remote Key Management Service provides key and associated certificate management, and remote signing service. KMSs are outside Notation trust boundary.
+- **OCSP Responder**: The [Online Certificate Status Protocol (OCSP)](https://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol) is an Internet protocol used for obtaining the revocation status of an X.509 digital certificate. Notation uses OCSP to validate certificate revocation status from OCSP Responder.
+
+## Notation sign artifacts using remote key
+
+### Data flow
+
+A signer uses Notation to sign artifacts with the private key stored in a KMS, and store signatures in the file system or in a registry.
+
+The following diagram illustrates Notation signing artifacts stored in registries with keys stored in KMS:
+
+![image](../media/notation-sign-remote.svg)
+
+The following diagram illustrates Notation signing artifacts as OCI image layout with keys stored in KMS:
+
+![image](../media/notation-sign-local.svg)
+
+According to [Notary signature specification](../specs/signature-specification.md), the signing payload is a JSON document which requires the [OCI descriptor](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/descriptor.md) of the artifact. For artifacts stored in a registry, the descriptor info can be retrieved by inspecting the [image manifest](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md) of the artifact. For artifacts stored as an [OCI Image layout](https://github.com/opencontainers/image-spec/blob/v1.0.0/image-layout.md), the descriptor info can be retrieved from the `index.json` file from OCI Image layout.
```

Comment:
Just a question: do you want to link the latest version of image-spec?

Reply:
No. I want to use stable version for image-layout and descriptor. I will update the version of descriptor.md to v1.0.0 as well.